### ISSUE TITLE
0.59.2 alpha in a vm proxmox with pci passthrough 4 x i225-lm (igc) reboot unexpected

### ISSUE THREAD

**Issue by Kalimeiro** [Ysurac/openmptcprouter#2631]
## Expected Behavior

OMR works stable and reliable.

## Current Behavior

OMR suddenly and randomly reboot, it restart after few minutes or few hours. it can be after 2 minutes as in 2 hours or in 10 hours...

VM console does not display any kernel panic, it suddenly reboot.

## Possible Solution

No idea how to debug.

## Specifications

  - OpenMPTCProuter version: 0.59.2 alpha
  - OpenMPTCProuter VPS version: 0.1028
  - OpenMPTCProuter VPS provider: OVH
  - OpenMPTCProuter platform: x86_64
  - Country: FR

How can i log dmesg kernel and save it in a file before reboot to understand what's happen ?

**Comment by Kalimeiro** [comment]
i'm trying same version with 5.15 kernel.

**Comment by Kalimeiro** [comment]
with kernel 5.15, no reboot unexpected.

kmod-igc on 5.4 kernel has bug like :

- force speed (auto negociation not working)
- wrong typo on scale value
- infinite loop
- ...

how can i try to compile 5.4 kernel with 5.15 igc driver ?

**Comment by Kalimeiro** [comment]
Ok for information we reproduced the same bug with 5.15 kernel after many day of good works.

we detected a problem with SeaBios on hypervisor when using pci passthrough, it reset and restoring BARs lot of time when starting the VM and after VM initialized and working.

we change to UEFI OVMF Bios and change OMR to EFI version, no restoring BARs found and OMR is working great with high workload in 7 hours.
